distributed/scheduler.py

```diff
@@ -2860,18 +2894,19 @@ def decide_worker(dependencies, stacks, processing, who_has, has_what, restricti
     if len(workers) == 1:
         return first(workers)
 
-    commbytes = {w: sum([nbytes.get(k, 1000) for k in dependencies[key]
-                                   if w not in who_has[k]])
-                 for w in workers}
-
-    minbytes = min(commbytes.values())
-    workers = {w for w, nb in commbytes.items() if nb == minbytes}
-
-    def objective(w):
-        return (len(stacks[w]) + len(processing[w]),
-                len(has_what.get(w, ())))
-    worker = min(workers, key=objective)
-    return worker
+    # Select worker that will finish task first
+    best_time = 1e100
+    best_worker = None
+    for w in workers:
+        comm_bytes = sum([nbytes.get(k, 1000) for k in dependencies[key]
+                          if w not in who_has[k]])
+        stack_time = stack_duration[w] / ncores[w]
```

Comment:
This is the actual usage for the durations - but the objective function changed too. Is the only effective change here that we measure the execution time of the stack instead of the number of tasks?